drivers/pinmux/pinmux_npcx.c

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2020 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <soc.h>
+#include <drivers/pinmux.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(pimux_npcx, LOG_LEVEL_ERR);
+
+/* Driver config */
+struct npcx_pinctrl_config {
+	/* scfg controller base address */
+	scfg_inst_t base;
+};
+
+/* Default io list which default functionality are not IOs */
+#define DT_DRV_COMPAT nuvoton_npcx_pinctrl_def
+static const struct npcx_alt def_alts[] = DT_NPCX_ALT_ITEMS_LIST(0);
+
+static const struct npcx_pinctrl_config npcx_pinctrl_cfg = {
+	.base = (scfg_inst_t)DT_REG_ADDR(DT_NODELABEL(scfg)),
+};
+
+/* Driver convenience defines */
+#define DRV_CONFIG(dev) \
+	((const struct npcx_pinctrl_config *)(dev)->config_info)
+
+#define HAL_INSTANCE(dev) \
+	(scfg_inst_t)(DRV_CONFIG(dev)->base)
+
+/******************************************************************************/
+/* Pin-control local functions */
+static void npcx_pinctrl_alt_sel(const struct npcx_alt *alt, int alt_func)
+{
+	scfg_inst_t inst = npcx_pinctrl_cfg.base;
+	uint8_t alt_mask = 1 << alt->bit;
+
+	/*
+	 * alt_func = 0 -> GPIO functionality
+	 * alt->inverted -> Set DEVALT bit for GPIO
+	 */
+	if (alt_func ^ alt->inverted)
+		NPCX_DEVALT(inst, alt->group) |=  alt_mask;
+	else
+		NPCX_DEVALT(inst, alt->group) &= ~alt_mask;
+}
+
+/******************************************************************************/
+/* Soc specific pin-control functions */
+void soc_pinctrl_mux_configure(const struct npcx_alt *alts_list,
+		      uint8_t alts_size, int altfunc)
+{
+	int i;
+
+	for (i = 0; i < alts_size; i++, alts_list++) {
+		npcx_pinctrl_alt_sel(alts_list, altfunc);
+	}
+}
+
+/******************************************************************************/
+/* Pin-control driver registration */
+static int npcx_pinctrl_init(struct device *dev)
+{
+	scfg_inst_t inst = HAL_INSTANCE(dev);
+
+#ifdef CONFIG_SOC_SERIES_NPCX7
+	/*
+	 * Set bit 7 of DEVCNT again for npcx7 series. Please see Errata
+	 * for more information. It will be fixed in next chip.
+	 */
+	SET_BIT(NPCX_DEVCNT(inst), 7);
+#endif
+
+	/* Lock VCC_RST# alternative bit in case switch to GPO77 unexpectedly */
```

Comment:
Can you expand that? It makes no sense to me

Reply:
Done